checked_message.go

```diff
@@ -41,6 +47,25 @@ type CheckedMessage struct {
 // NewCheckedMessage constructs a CheckedMessage. It's only intended for use by
 // wrapper libraries, and shouldn't be necessary in application code.
 func NewCheckedMessage(logger Logger, lvl Level, msg string) *CheckedMessage {
+	for {
+		val := cmPool.Get()
+		if val == nil {
+			break
+		}
```

Comment:
I don't think that this nil check is necessary (once we have a `New` on `cmPool`); we never return nil CMs to the pool, and we should be able to vigorously exercise these code paths in tests.